src/NServiceBus.Core/Unicast/Transport/TransportReceiver.cs

```diff
@@ -102,11 +93,11 @@ public virtual void Start()
                 throw new InvalidOperationException("The transport is already started");
             }
 
-            InitializePerformanceCounters(dequeueSettings.QueueName);
-
             Logger.DebugFormat("Pipeline {0} is starting receiver for queue {0}.", Id, dequeueSettings.QueueName);
 
-            receiver.Init(dequeueSettings);
+            var dequeueInfo = receiver.Init(dequeueSettings);
+            pipeline.Initialize(new PipelineInfo(id, dequeueInfo.PublicAddress));
+            pipeline.Warmup();
```

Comment:
I'd say push the task as high up in the call stack as possible. So make `pipeline.Warmup();` return a task as well and synchronize here or even push it up more